### ISSUE TITLE
API docs sidebar has undesired entries

### ISSUE THREAD

**Issue by mit-mit** [flutter/flutter#15290]
The sidebar / navigation menu on https://docs.flutter.io/ starts with two nice, overall categories 'FLUTTER' and 'DART CORE', but then after that lists a whole bunch of smaller categories ('ASYNC', 'BOOLEAN_SELECTOR', etc.). Are those not all part of the larger categories, and thus can be excluded?

**Comment by mit-mit** [comment]
cc @jcollins-g

**Comment by jcollins-g** [comment]
The categories are not actually categories internally to dartdoc, they are packages.  So we have the flutter package, Dart Core (a virtual package for the SDK), and all other packages.

Do you want to stop documenting the other packages, or to change how they are appearing?

**Comment by mit-mit** [comment]
Maybe I'm just confused. What's the difference between these two?
https://docs.flutter.io/flutter/dart-async/dart-async-library.html
https://docs.flutter.io/flutter/package-async_async/package-async_async-library.html

Are they really these two?
https://api.dartlang.org/stable/1.24.3/dart-async/dart-async-library.html
https://pub.dartlang.org/packages/async

**Comment by jcollins-g** [comment]
Correct.  It would appear that in that case, the async package is a package intended to extend `dart:async` with useful utility classes.  I'm guessing that something similar is going on in the case of other apparent duplicates -- you'll notice that the two have different documentation, different member classes, and so on.

This _is_ rather confusing though.  Haven't thought much on how to improve it yet.

**Comment by mit-mit** [comment]
What determines which set of packages we include on https://docs.flutter.io/ ?

**Comment by jcollins-g** [comment]
This is calculated via the transitive set of all package dependencies beginning at this root:  https://github.com/flutter/flutter/blob/master/dev/tools/pubspec.yaml,

minus all packages listed here:

https://github.com/flutter/flutter/blob/c0118ea4b98423358625a570e4aa9f8bb492a7a2/dev/tools/dartdoc.dart#L130

I [wrote this line](https://github.com/flutter/flutter/pull/13432) when I [made it possible to configure which packages show up here](https://github.com/dart-lang/dartdoc/pull/1524), but I did so in a way that preserved the old hacky behavior it replaced rather than doing something new.  You should be able to just add package names to that list to hide them, or remove them to unhide them.

**Comment by mit-mit** [comment]
I wonder if we should simplify things and just show the direct dependencies listed in [that pubspec](https://github.com/flutter/flutter/blob/master/dev/tools/pubspec.yaml)? 

cc @devoncarew @Hixie WDYT?

**Comment by mit-mit** [comment]
cc @devoncarew @Hixie WDYT?

**Comment by devoncarew** [comment]
My preference would be to just document packages that were intended to be public API. I think that means the public packages in `//packages/` - `flutter`, `flutter_driver`, and `flutter_test` - and the libraries that those explicitly export - `meta`, dart:ui`, ... . 

I think @Hixie prefers that we doc all possible reachable libraries. I do think this then presents a lot of information for users to wade through. Perhaps @InMatrix has thoughts here.

**Comment by InMatrix** [comment]
I don't know enough to state a preference. It might be the case that framework engineers need a more complete set of API docs than app developers. If that's true, we can potentially put non-public API as Devon suggested into an "advanced" or "internal dependencies" category and have it show up after more commonly used packages.

**Comment by Hixie** [comment]
I think we should document everything that the various APIs we expose themselves expose. The real problem is the `test` package, which depends on everything in the world (literally all of the following come from `test`: async, shelf_static, web_socket_channel, meta, matcher, io, shelf_web_socket, term_glyph, multi_server_socket, stream_channel, boolean_selector, shelf, source_map_stack_trace, source_span, string_scanner, js, vm_service_client, source_maps, package_resolver, args, collection, typed_data, pool, glob, stack_trace, pub_semver, path, shelf_packages_handler, yaml, http_multi_server, node_preamble, analyzer).

**Comment by goderbauer** [comment]
How should we proceed here? There doesn't seem to be anything actionable left here if we go with what @Hixie said in his last comment.

**Comment by zoechi** [comment]
The dependency on `test` was removed #24007 and the list in the sidebar is much shorter now.
